src/Domain/EmbeddingMaps/AffineMap.hpp

```diff
@@ -0,0 +1,97 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines the class AffineMap.
+
+#pragma once
+
+#include <memory>
+
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Domain/EmbeddingMaps/EmbeddingMap.hpp"
+#include "Parallel/CharmPupable.hpp"
+
+namespace EmbeddingMaps {
+
+/// Linear map from \f$\xi \in [A, B]\rightarrow x \in [a, b]\f$.
+/// The formula for the mapping is...
+/*! \f[
+  x = \frac{b}{B-A} (\xi-A) +\frac{a}{B-A}(B-\xi)
+ \f]
+ \f[
+  \xi =\frac{B}{b-a} (x-a) +\frac{A}{b-a}(b-x)
+ \f]
+*/
+class AffineMap : public EmbeddingMap<1, 1> {
+ public:
+  AffineMap(const double A, const double B, const double a, const double b)
+      : A_(A), B_(B), a_(a), b_(b), L_(B - A), l_(b - a) {}
+
+  AffineMap() = default;
+  ~AffineMap() override = default;
+  AffineMap(const AffineMap&) = delete;
+  AffineMap(AffineMap&&) noexcept = default;  // NOLINT
+  AffineMap& operator=(const AffineMap&) = delete;
+  AffineMap& operator=(AffineMap&&) = default;
+
+  std::unique_ptr<EmbeddingMap<1, 1>> get_clone() const override {
+    return std::make_unique<AffineMap>(A_, B_, a_, b_);
+  }
+
+  Point<1, Frame::Grid> operator()(
+      const Point<1, Frame::Logical>& xi) const override {
+    return Point<1, Frame::Grid>((l_ * xi[0] + a_ * B_ - b_ * A_) / L_);
+  }
+
+  Point<1, Frame::Logical> inverse(
+      const Point<1, Frame::Grid>& x) const override {
+    return Point<1, Frame::Logical>((L_ * x[0] - a_ * B_ + b_ * A_) / l_);
+  }
+
+  double jacobian(const Point<1, Frame::Logical>& /* xi */, size_t ud,
+                  size_t ld) const override {
+    ASSERT(0 == ld, "ld = " << ld);
+    ASSERT(0 == ud, "ud = " << ud);
+    return l_ / L_;
```

Comment:
Since the jacobians are constant in this map should we have those stored as well? It would save us having to recompute them.